Add example on defining Clerk keys per tenant from `clerkMiddleware` (#1508)
clerk/clerk-docs: docs/references/nextjs/clerk-middleware.mdx

openssl rand --hex 32
```

For multi-tenant applications, you can dynamically define Clerk keys depending on the incoming request. Here's an example:

```ts {{ filename: 'middleware.ts' }}
import { clerkMiddleware } from '@clerk/nextjs/server'

 const tenantKeys = {
  tenant1: { publishableKey: 'pk_tenant1...', secretKey: 'sk_tenant1...' },
  tenant2: { publishableKey: 'pk_tenant2...', secretKey: 'sk_tenant2...' },
};

export default clerkMiddleware(
  (auth, req) => {
    // Add your middleware checks
  },
  (req) => {

>     // Resolve tenant from request

Author: ```suggestion
    // Resolve tenant based on the request
```

openssl rand --hex 32
```

> For multi-tenant applications, you can dynamically define Clerk keys depending on the incoming request. Here's an example:

Reviewer: ```suggestion
For multi-tenant applications, you can dynamically define Clerk keys depending on the incoming request, as shown in the following example:
```

```ts {{ filename: 'middleware.ts' }}
import { clerkMiddleware } from '@clerk/nextjs/server'

const tenantKeys = {
  tenant1: { publishableKey: 'pk_tenant1...', secretKey: 'sk_tenant1...' },
  tenant2: { publishableKey: 'pk_tenant2...', secretKey: 'sk_tenant2...' },
}

export default clerkMiddleware(
  (auth, req) => {
    // Add your middleware checks
  },
  (req) => {
    // Resolve tenant based on the request
    const tenant = getTenant(req)

>     // You would typically fetch these keys from a external store or environment variables.

Reviewer: Should this comment go above the `const tenantKeys = {` line ?

Author: Yeah, I think it makes more sense to move it there.
